### PR TITLE
enhanced routing (changes from #70)

### PR DESCRIPTION
**Pull request by paulheinr**
This PR includes changes of the router from #70. Additionally changes for #85 and #84.

### CODE REVIEW

**Comment by paulheinr** [comment]
@Janekdererste can you have a look at the changes from 4th commit? https://github.com/Janekdererste/rust_q_sim/pull/89/commits/522501e75ad1230537a225af6646ecb0abfa0ff6 I refactored the network converter and adapted the controller. Other things should be straight forward.

File: src/simulation/messaging/messages.rs [hunk 1]
```diff
@@ -172,6 +173,18 @@ impl Agent {
         routing_mode: RoutingMode,
     ) -> Agent {
         let plan = Plan::from_io(io_person.selected_plan(), id_mappings, routing_mode);
+
+        if plan.acts.is_empty() {
```
Comment (Janekdererste) [inline src/simulation/messaging/messages.rs:177]: does this happen? This should not happen. Agents must have at least one activity.

File: src/simulation/messaging/messages.rs [hunk 2]
```diff
@@ -172,6 +173,18 @@ impl Agent {
...
+            debug!("There is an empty plan for person {:?}", io_person.id);
+        }
+
+        if plan.acts.len() == 1 {
```
Comment (Janekdererste) [inline src/simulation/messaging/messages.rs:181]: This is ok. These are stay home agents.

File: src/simulation/messaging/messages.rs [hunk 3]
```diff
@@ -422,6 +435,21 @@ impl Plan {
     }
 
     fn get_full_plan_for_routing(io_plan: &IOPlan, id_mappings: &MatsimIdMappings) -> Plan {
+        if io_plan.elements.is_empty() {
```
Comment (Janekdererste) [inline src/simulation/messaging/messages.rs:438]: One more reason to move the routing logic somewhere else. It is getting more and more messy.

**Review by Janekdererste — APPROVED** [review]
I guess this works for now.

**Pull request merged**